cpp/src/io/copc_writer_public.cpp

```diff
@@ -59,6 +59,7 @@ void Writer::InitWriter(std::ostream &out_stream, const CopcConfigWriter &copc_c
         // If not we use the copc_config_writer provided
         this->config_ = std::make_shared<CopcConfigWriter>(copc_config_writer);
     }
+    config_->LasHeader()->IsCopc(true);
```

Comment:
I feel like this should be handled within CopcConfig constructors? So by the time it gets here we are sure it's true?

Reply:
Like `copc_config.hpp` line 25 and 29 as part of the constructors

Reply:
The problem where I ran into it was in copcconfig copy construcotrs, which is where them ain problem lies anyways, i.e. taking a CopcConfig and copying it into a LazConfigWriter using the copy constructor

Reply:
I'm not sure I understand the issue? CopcConfig what have the flag to `True`, then the LazConfigWriter would also have it to `True`, then when if you create a LazWriter from that LazConfigWriter, it would set it to false within GenerateConfig?